timeseries/src/autogluon/timeseries/utils/features.py

```diff
@@ -183,7 +194,16 @@ def transform(self, data: TimeSeriesDataFrame, data_frame_name: str = "data") ->
         else:
             static_features = None
 
-        return TimeSeriesDataFrame(pd.concat(dfs, axis=1), static_features=static_features)
+        ts_df = TimeSeriesDataFrame(pd.concat(dfs, axis=1), static_features=static_features)
+
+        covariates_names = self.covariate_metadata.covariates
+        if len(covariates_names) > 0:
+            ts_df[covariates_names] = ts_df[covariates_names].fill_missing_values()
+            # If some series consist completely of NaNs, fill with median
```

Comment:
The first line 
```python
ts_df[covariates_names] = ts_df[covariates_names].fill_missing_values()
```
does ffill + bfill for all covariates (per item). If there are still missing values left after that, it means that for some time series that covariate consists completely of NaNs. In this case, we use the median of the covariate in the dataset for imputation.

Thinking about it, we probably need to store the median covariate values of the training data during fit and use those for imputation. This would make more sense in case user tries to predict items one by one (and for some items the covariate is all-nan)

Reply:
That's quite complicated logic. If we can get away with filling zeros for this edge case for example, I'm OK with that (with a loud warning as well) just to save complexity.